A few `mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test` test fail when `lxml` is built against libxml2-2.15

**Bug Report**

Newer libxml2 versions changed the HTML output to use `<meta charset="UTF-8">` instead of the long form. This causes a bunch of mismatches when running tests against `lxml` built locally (i.e. on Linux distributions), such as:

```pytb
============================================================== FAILURES ===============================================================
______________________________________________________ testSpecialAnyHtmlReport _______________________________________________________
[gw0] linux -- Python 3.13.9 /tmp/mypy/.venv/bin/python3
data: /tmp/mypy/test-data/unit/reports.test:477:
Failed: Output file tmp/report/html/n.py.html did not match its expected output
-------------------------------------------------------- Captured stderr call ---------------------------------------------------------
Expected:
  <html>
  <head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> (diff)
  <link rel="stylesheet" type="text/css" href="../mypy-html.css">
  </head>
  <body>
  ...
  </table>
  </body>
  </html>
Actual:
  <html>
  <head>
  <meta charset="UTF-8"> (diff)
  <link rel="stylesheet" type="text/css" href="../mypy-html.css">
  </head>
  <body>
  ...
  </table>
  </body>
  </html>

Alignment of first line difference:
  E: <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  A: <meta charset="UTF-8">
           ^
Update the test output using --update-data (implies -n0; you can additionally use the -k selector to update only specific tests)
```

The full list of failures is:

```
FAILED mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test::testUnreachableCodeMarkedAsAny
FAILED mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test::testClassDefIsNotTreatedAsEmpty
FAILED mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test::testTypeVarTreatedAsEmptyLine
FAILED mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test::testHtmlReportMemberExprNoUnanalyzed
FAILED mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test::testSpecialAnyHtmlReport
FAILED mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test::testHtmlReportOnNamespacePackagesWithExplicitBases
```

Could the test be made more resilient to this difference, please?

**To Reproduce**

```
# install libxml2 >= 2.15, e.g.:
pacman -Syu libxml2 libxslt base-devel
uv venv -p 3.13
uv pip install --no-binary lxml -e . -r test-requirements.txt
.venv/bin/pytest mypy/test/testcmdline.py::PythonCmdlineSuite::reports.test
```

**Expected Behavior**

Tests passing.

**Actual Behavior**

Failures as noted above.

**Your Environment**

- Mypy version used: 1.18.2 and 2c6c3959356674262d9b2c2dc43a33486e807a9c
- Python version used: 3.13.9

--- a/mypy/test/helpers.py
+++ b/mypy/test/helpers.py
@@ -444,6 +444,8 @@ def check_test_output_files(
             if testcase.suite.native_sep and os.path.sep == "\\":
                 normalized_output = [fix_cobertura_filename(line) for line in normalized_output]
             normalized_output = normalize_error_messages(normalized_output)
+        if os.path.basename(testcase.file) == "reports.test":
+            normalized_output = normalize_report_meta(normalized_output)
         assert_string_arrays_equal(
             expected_content.splitlines(),
             normalized_output,
@@ -465,6 +467,13 @@ def normalize_file_output(content: list[str], current_abs_path: str) -> list[str
     result = [re.sub(r"\b" + re.escape(base_version) + r"\b", "$VERSION", x) for x in result]
     result = [timestamp_regex.sub("$TIMESTAMP", x) for x in result]
     return result
+
+
+def normalize_report_meta(content: list[str]) -> list[str]:
+    # libxml 2.15 and newer emits the "modern" version of this <meta> element.
+    # Normalize the old style to look the same.
+    html_meta = '<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">'
+    return ['<meta charset="UTF-8">' if x == html_meta else x for x in content]
 
 
 def find_test_files(pattern: str, exclude: list[str] | None = None) -> list[str]:

--- a/test-data/unit/reports.test
+++ b/test-data/unit/reports.test
@@ -118,7 +118,7 @@ class A(object):
 [outfile report/html/n.py.html]
 <html>
 <head>
-<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<meta charset="UTF-8">
 <link rel="stylesheet" type="text/css" href="../mypy-html.css">
 </head>
 <body>
@@ -172,7 +172,7 @@ T = TypeVar('T')
 [outfile report/html/n.py.html]
 <html>
 <head>
-<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<meta charset="UTF-8">
 <link rel="stylesheet" type="text/css" href="../mypy-html.css">
 </head>
 <body>
@@ -214,7 +214,7 @@ def bar(x):
 [outfile report/html/n.py.html]
 <html>
 <head>
-<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<meta charset="UTF-8">
 <link rel="stylesheet" type="text/css" href="../mypy-html.css">
 </head>
 <body>
@@ -255,7 +255,7 @@ old_stdout = sys.stdout
 [outfile report/html/n.py.html]
 <html>
 <head>
-<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<meta charset="UTF-8">
 <link rel="stylesheet" type="text/css" href="../mypy-html.css">
 </head>
 <body>
@@ -487,7 +487,7 @@ DisplayToSource = Callable[[int], int]
 [outfile report/html/n.py.html]
 <html>
 <head>
-<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<meta charset="UTF-8">
 <link rel="stylesheet" type="text/css" href="../mypy-html.css">
 </head>
 <body>
@@ -529,7 +529,7 @@ namespace_packages = True
 [outfile report/html/folder/subfolder/something.py.html]
 <html>
 <head>
-<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<meta charset="UTF-8">
 <link rel="stylesheet" type="text/css" href="../../../mypy-html.css">
 </head>
 <body>

